packages/common/container-definitions/src/audience.ts

```diff
@@ -21,6 +20,23 @@ export interface IAudienceOwner extends IAudience {
 	 * @returns if a client was removed from the audience
 	 */
 	removeMember(clientId: string): boolean;
+
+	/**
+	 * Notifies Audience that current clientId has changed.
+	 * See {@link IAudience.currentClientId} and {@link IAudienceEvents}'s "clientIdChanged" event for more details.
+	 */
+	setCurrentClientId(clientId: string | undefined): void;
+}
+
+/**
+ * @public
+ */
+export interface IAudienceEvents extends IEvent {
+	// eslint-disable-next-line @typescript-eslint/prefer-function-type
+	(
+		event: "addMember" | "removeMember" | "clientIdChanged",
+		listener: (clientId: string, client: IClient) => void,
```

Comment:
These don't seem like the right arguments for a "clientIdChanged" event - I would expect it to be either emptyu or look more like:

```suggestion
		listener: (oldClientId: string) => void,
```

Reply:
All of our existing API surfaces notify about new clientID only. See raiseConnectedEvent() used at loader & runtime layers. I was simply following existing patterns. I think having both (old & new) and dropping IClient is the right way to go here...

Reply:
I've changed it to expose both old & new clientId